### PR TITLE
Refactor/add transcript api utils.

### PR DESCRIPTION
**Pull request by Qubad786**

### CODE REVIEW

**Comment by Qubad786** [comment]
@muhammad-ammar @mushtaqak @irfanuddinahmad please have a look.

File: edxval/api.py [hunk 1]
```diff
@@ -245,26 +261,23 @@ def get_video_transcript_data(video_ids, language_code):
...
+        A list containing transcript language codes for the Video.
     """
     available_languages = VideoTranscript.objects.filter(
-        video__edx_video_id__in=video_ids
+        video__edx_video_id=video_id
     ).values_list(
         'language_code', flat=True
     )
-    return list(set(available_languages))
+    return list(available_languages)
```
[the remark this answers is not in the document]
Comment (Qubad786) [reply edxval/api.py]: For some reason, the returned object was not serializable on passing it into the context while rendering a template. Exception that I got:
```bash
Traceback (most recent call last):
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 132, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/utils/decorators.py", line 145, in inner
    return func(*args, **kwargs)
  File "/edx/app/edxapp/edx-platform/common/djangoapps/util/json_request.py", line 53, in parse_json_into_request
    return view_function(request, *args, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/contrib/auth/decorators.py", line 22, in _wrapped_view
    return view_func(request, *args, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/views/decorators/http.py", line 45, in inner
    return func(request, *args, **kwargs)
  File "/edx/app/edxapp/edx-platform/cms/djangoapps/contentstore/views/videos.py", line 178, in videos_handler
    return videos_index_html(course)
  File "/edx/app/edxapp/edx-platform/cms/djangoapps/contentstore/views/videos.py", line 670, in videos_index_html
    return render_to_response('videos_index.html', context)
  File "/edx/app/edxapp/edx-platform/common/djangoapps/edxmako/shortcuts.py", line 169, in render_to_response
    return HttpResponse(render_to_string(template_name, dictionary, namespace, request), **kwargs)
  File "/edx/app/edxapp/edx-platform/common/djangoapps/edxmako/shortcuts.py", line 159, in render_to_string
    return template.render(dictionary, request)
  File "/edx/app/edxapp/edx-platform/common/djangoapps/edxmako/template.py", line 59, in render
    return self.mako_template.render_unicode(**context_dictionary)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/mako/template.py", line 454, in render_unicode
    as_unicode=True)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/mako/runtime.py", line 829, in _render
    **_kwargs_for_callable(callable_, data))
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/mako/runtime.py", line 864, in _render_context
    _exec_template(inherit, lclcontext, args=args, kwargs=kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/mako/runtime.py", line 890, in _exec_template
    callable_(context, *args, **kwargs)
  File "/tmp/mako_cms/5ad50d53547dca6408eb97d1b0794e5f/base.html.py", line 252, in render_body
    context['self'].content(**pageargs)
  File "/tmp/mako_cms/5ad50d53547dca6408eb97d1b0794e5f/videos_index.html.py", line 161, in render_content
    __M_writer(filters.html_escape(filters.decode.utf8(json.dumps(previous_uploads, cls=DjangoJSONEncoder))))
  File "/usr/lib/python2.7/json/__init__.py", line 251, in dumps
    sort_keys=sort_keys, **kw).encode(obj)
  File "/usr/lib/python2.7/json/encoder.py", line 207, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/usr/lib/python2.7/json/encoder.py", line 270, in iterencode
    return _iterencode(o, 0)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/serializers/json.py", line 115, in default
    return super(DjangoJSONEncoder, self).default(o)
  File "/usr/lib/python2.7/json/encoder.py", line 184, in default
    raise TypeError(repr(o) + " is not JSON serializable")
TypeError: [] is not JSON serializable
```

File: edxval/models.py [hunk 2]
```diff
@@ -122,6 +122,18 @@ def get_absolute_url(self):
...
+    def get_or_none(cls, **filter_kwargs):
+        """
+        Returns a video or None.
+        """
+        try:
+            video = cls.objects.get(**filter_kwargs)
+        except cls.DoesNotExist:
+            video = None
+
+        return video
```
[the remark this answers is not in the document]
Comment (Qubad786) [reply edxval/models.py]: settled: `get_or_none ` should fail if someone is expecting multiple matching videos from it.

File: edxval/api.py [hunk 3]
```diff
@@ -280,6 +293,28 @@ def get_video_transcript_url(video_id, language_code):
         return video_transcript.url()
 
 
+def create_video_transcript(video_id, language_code, file_format, content, provider=TranscriptProviderType.CUSTOM):
```
Comment (mushtaqak) [inline edxval/api.py]: Where are we using this method?

Comment (mushtaqak) [reply edxval/api.py]: Why are we not using `create_or_update_video_transcript` method?

Comment (Qubad786) [reply edxval/api.py]: This is required for migration task. we cannot use `create_or_update` as it also overwrites the existing transcript.

File: edxval/api.py [hunk 4]
```diff
@@ -39,6 +40,13 @@ class SortDirection(Enum):
     desc = "desc"
 
 
+def generate_video_id():
```
[the remark this answers is not in the document]
Comment (Qubad786) [reply edxval/api.py]: I plan to use it in `videos.py` too!

**Pull request merged**